spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java

```diff
@@ -1473,9 +1503,371 @@ static void queryWithQueryOptions(DatabaseClient dbClient) {
+        SpannerException se = (SpannerException) e.getCause();
+        if (se.getErrorCode() == ErrorCode.FAILED_PRECONDITION
+            && se.getMessage()
+                .contains("Please retry the operation once the pending restores complete")) {
+          restoreAttempts++;
+          if (restoreAttempts == 10) {
+            System.out.println(
+                "Restore operation failed 10 times because of other pending restores. "
+                + "Giving up restore.");
+            return;
+          }
+          // Wait and then retry.
+          System.out.println(String.format(
+              "Restoring backup %s to database %s must wait because of other pending restore",
+              backup.getId().getBackup(),
+              restoreToDatabase.getDatabase()));
+          Uninterruptibles.sleepUninterruptibly(60L, TimeUnit.SECONDS);
+        } else {
+          throw se;
+        }
+      } catch (InterruptedException e) {
+        throw SpannerExceptionFactory.propagateInterrupt(e);
+      }
+    }
+  }
+  // [END spanner_restore_backup]
+
+  // [START spanner_update_backup]
+  static void updateBackup(DatabaseAdminClient dbAdminClient, DatabaseId databaseId) {
+    // Get the most recent sample backup of the sample database.
+    String backupName =
+        String.format(
+            "%s_%02d",
+            databaseId.getDatabase(), LocalDate.now().get(ChronoField.ALIGNED_WEEK_OF_YEAR));
+    BackupId backupId = BackupId.of(databaseId.getInstanceId(), backupName);
+    // Get current backup metadata.
+    Backup backup = dbAdminClient.newBackupBuilder(backupId).build().reload();
+    // Add 30 days to the expire time.
+    // Expire time must be within 366 days of the create time of the backup.
+    Timestamp expireTime =
+        Timestamp.ofTimeMicroseconds(
+            TimeUnit.SECONDS.toMicros(backup.getExpireTime().getSeconds())
+                + TimeUnit.NANOSECONDS.toMicros(backup.getExpireTime().getNanos())
+                + TimeUnit.DAYS.toMicros(30L));
+    System.out.println(String.format(
+        "Updating expire time of backup [%s] to %s...",
+        backupId.toString(),
+        LocalDateTime.ofEpochSecond(
+            expireTime.getSeconds(),
+            expireTime.getNanos(),
+            OffsetDateTime.now().getOffset()).toString()));
+
+    // Update expire time.
+    backup = backup.toBuilder().setExpireTime(expireTime).build();
+    backup.updateExpireTime();
+    System.out.println("Updated backup [" + backupId + "]");
+  }
+  // [END spanner_update_backup]
+
+  // [START spanner_delete_backup]
+  static void deleteBackup(DatabaseAdminClient dbAdminClient, DatabaseId databaseId) {
+    // Get the most recent sample backup of the sample database.
+    String backupName =
+        String.format(
+            "%s_%02d",
+            databaseId.getDatabase(), LocalDate.now().get(ChronoField.ALIGNED_WEEK_OF_YEAR));
+    BackupId backupId = BackupId.of(databaseId.getInstanceId(), backupName);
```

Comment:
Same here regarding getting the backupId as in input.

Reply:
done.